transcom/mymove
# B-21667 B-21374 I-13550 INT

## [Agility Issue](https://www13.v1host.com/USTRANSCOM38/Issue.mvc/Summary?oidToken=Issue%3A1034058)
## [Agility ticket B-21667](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21667)
## [Agility ticket B-21374](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21374)

## Summary

B-21374 fix
A mysterious 0 value was appearing on the allowance details in the case of 0 dependents under 12 and 0 over 12. Issue was written up for it, fix was just adding a > 0 for the each of the checks for info?.dependentsTwelveAndOver || info?.dependentsUnderTwelve in src/components/Office/DefinitionLists/AllowancesList.jsx.

B-21667 fix
As part of troubleshooting the above, noticed that the Unaccompanied Baggage Allowance field was not appearing on the Allowance section. Needed to update ubAllowance to unaccompaniedBaggageAllowance, after changing that variable as part of [B-21487](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21487), which I have merged into my B-21667 branch.

### How to test

1. Login as a customer, creating a UB shipment with the UB FF turned on, selecting no dependents
2. Log in as an SC and view the Allowances section for the move.
3. Check that the DB has 0 values for the dependents under 12 and dependents 12 and over.
4. Verify that you do not see the 0 just hanging out in the dependents section
Should _NOT_ see this 0:
![image](https://github.com/user-attachments/assets/ae3a5fdf-2892-4333-8616-a3906f2e7434)
5. Check that you have a non-zero value in the UB allowance column of the order entitlement
6. Verify that you see the Unaccompanied Baggage Allowance field in the Allowances section with the UB allowance value.
![image](https://github.com/user-attachments/assets/12bd02e0-8c31-4347-ab7f-10b3fd73ad5b)

## Try to break
1. Play around with 0 values for dependents under 12, dependents 12 and over, and ub allowance. You should see 0 values handled by either not being shown or shown with a -, and non-zero values displayed correctly
